src/macaron/repo_finder/provenance_extractor.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains methods for extracting repository and commit metadata from provenance files."""
+import logging
+from typing import TypeVar
+
+from macaron.errors import MacaronError
+from macaron.slsa_analyzer.provenance.intoto import InTotoPayload, InTotoV1Payload, InTotoV01Payload
+from macaron.util import JsonType
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ProvenanceExtractionException(MacaronError):
+    """When there is an error while extracting from provenance."""
+
+
+SLSA_V01_DIGEST_SET_ALGORITHMS = ["sha1"]
+SLSA_V02_DIGEST_SET_ALGORITHMS = ["sha1"]
+SLSA_V1_DIGEST_SET_ALGORITHMS = ["sha1", "gitCommit"]
```

Comment:
The name should mention "commit hash" because these hash algorithms are specifically for git commit hash and not just any general digest set value.